### ISSUE TITLE
Treinar o word2vec levando em conta max_seq_length e a dimensão do embedding (embedding_dim)

### ISSUE THREAD

**Issue by HaraHeique** [HaraHeique/TCC-rede-neural-siamesa#29]
- Pesquisar word2vec em língua inglesa mais utilizados ou mais promissores;
- Pesquisar se existe word2vec específico para textos literários da língua inglesa.

**Comment by HaraHeique** [comment]
Fidelis definiu o seguinte através de mensagens:

- Verificar qual word2vec a gente está usando e migrar pra melhor e mais completa e mais atual versão que usa a Wikipédia.;
- Verificar os modelos pré-treinados BERT e fazer o download do mais atual e completo.

[Modelo pré-treinado do word embedding do wikipédia](http://vectors.nlpl.eu/explore/embeddings/en/models/)
[Modelo pré-treinado do word embedding BERT](https://huggingface.co/transformers/pretrained_models.html)

**Comment by HaraHeique** [comment]
@Fideliszan  Atualmente estamos utilizando o [word2vec da google news](https://drive.google.com/file/d/0B7XkCwpI5KDYNlNUTTlSS21pQmM/edit?resourcekey=0-wjGZdNAUop6WykTtMip30g). Mas podemos modificar para outro word embedding mais específico, mas caso mudamos e temos uma melhor performance será necessário mudar algumas coisas na parte escrita do cap. 3.
Quando mudarmos os word embeddings será necessário (talvez no caso) refatorar o código de criação da matriz de incorporação (embedding matrix) que serve como peso na rede neural além de guardar os embeddings correspondentes para cada palavra existente nos datasets (treinamento e predição).
Além disto vou testar a rede neural com os novos datasets que criei e colocar os resultados nesta issue.

**Comment by Fideliszan** [comment]
@HaraHeique, verdade. Se migramos para o word embeddings BERT precisaremos escrever uma seção falando sobre ele. A minha proposta inicial é migramos definitivamente para o word2vec treinado a partir da Wikipedia. Isso não trará grandes modificações ao texto. Teremos de alterar o código na parte da criação da matriz de incorporação e no armazenamento dos pesos, mas essa modificação não é difícil de fazer. Basta mudar o arquivo bin e colocar o word2vec pra rodar de novo.

Considere a migração para o BERT apenas se a diferença de performance dos modelos for muito relevante. Isso descobriremos com testes.
